cli/command/formatter/container_test.go

```diff
@@ -323,3 +324,37 @@ func TestContainerContextWriteWithNoContainers(t *testing.T) {
 		out.Reset()
 	}
 }
+
+func TestContainerContextWriteJSON(t *testing.T) {
+	containers := []types.Container{
+		{ID: "containerID1", Names: []string{"/foobar_baz"}, Image: "ubuntu"},
+		{ID: "containerID2", Names: []string{"/foobar_bar"}, Image: "ubuntu"},
+	}
+	out := bytes.NewBufferString("")
+	err := ContainerWrite(Context{Format: "{{json .}}", Output: out}, containers)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for i, line := range strings.Split(strings.TrimSpace(out.String()), "\n") {
+		t.Logf("Output: line %d: %s", i, out.String())
+		var m map[string]interface{}
+		if err := json.Unmarshal([]byte(line), &m); err != nil {
+			t.Fatal(err)
+		}
+		ident, ok := m["ID"]
+		if !ok {
+			t.Fatalf("field \"ID\" not found in %v", m)
+		}
+		names, ok := m["Names"]
+		if !ok {
+			t.Fatalf("field \"Names\" not found in %v", m)
+		}
+		if ident == "containerID1" {
+			assert.Equal(t, names, "foobar_baz")
+		} else if ident == "containerID2" {
+			assert.Equal(t, names, "foobar_bar")
+		} else {
+			t.Fatalf("expect containerID1 or containerID2, got %v", ident)
+		}
+	}
```

Comment:
These assertions aren't as strict as they could be, and I find them a little involved to read.

Would it be possible to just define the expected map[string]string and compare them directly with DeepEquals? A new `assert.DeepEquals()` would be a great addition to `pkg/testutil/assert`